e2e/fixtures/styled-components/src/server-html/stream.ts

```diff
@@ -0,0 +1,60 @@
+const encoder = new TextEncoder();
+const decoder = new TextDecoder();
+
+const OVERLAP = '</head>'.length - 1;
+
+/**
+ * Creates a transform stream that inserts HTML content before </head>.
+ * Used for injecting CSS-in-JS styles during streaming SSR.
+ */
+export function createHeadInsertionTransformStream(
+  insert: () => Promise<string> | string,
+): TransformStream<Uint8Array, Uint8Array> {
+  let found = false;
+  let pending = '';
+  let toInsert = '';
+
+  return new TransformStream({
+    async transform(chunk, controller) {
+      const content = await insert();
+
+      if (found) {
+        if (content) {
+          controller.enqueue(encoder.encode(content));
+        }
+        controller.enqueue(chunk);
+        return;
+      }
+
+      if (content) {
+        toInsert += content;
+      }
+
+      const decoded = decoder.decode(chunk);
```

Comment:
binary `chunk` doesn't necessary contain full utf-8 boundary, in which case this causes garbled character.
`rsc-html-stream` uses `decode(chunk, { stream: true })` and read remaining by `decode()` within single `setTimeout` task https://github.com/devongovett/rsc-html-stream/blob/9b858445f4f5817470f373ae266dea04d5fcfac3/server.js#L18

But not sure maybe if this transformer comes after `rsc-html-stream`, then there might not be such chunks reaching here anymore.